qiita_pet/handlers/study_handlers/tests/test_listing_handlers.py

```diff
@@ -0,0 +1,439 @@
+            'shared':
+                '<a target="_blank" href="mailto:shared@foo.bar">Shared</a>',
+            'publication_doi':
+                '<a target="_blank" href="http://dx.doi.org/10.100/123456">'
+                '10.100/123456</a>, <a target="_blank" '
+                'href="http://dx.doi.org/10.100/7891011">10.100/7891011</a>',
+            'pmid': '<a target="_blank" href="http://www.ncbi.nlm.nih.gov'
+                    '/pubmed/7891011">7891011</a>, <a target="_blank" href='
+                    '"http://www.ncbi.nlm.nih.gov/pubmed/123456">123456</a>',
+            'pi': '<a target="_blank" href="mailto:PI_dude@foo.bar">'
+                  'PIDude</a>',
+            'proc_data_info': self.proc_data_exp
+        }
+        self.exp = [self.single_exp]
+        super(TestHelpers, self).setUp()
+
+    def test_get_shared_links_for_study(self):
+        obs = _get_shared_links_for_study(Study(1))
+        exp = '<a target="_blank" href="mailto:shared@foo.bar">Shared</a>'
+        self.assertEqual(obs, exp)
+
+    def test_build_single_study_info(self):
+        study_proc = {1: {'18S': [4, 5, 6]}}
+        proc_samples = {4: self.proc_data_exp[0]['samples'],
+                        5: self.proc_data_exp[1]['samples'],
+                        6: self.proc_data_exp[2]['samples']}
+        study_info = {
+            'study_id': 1,
+            'email': 'test@foo.bar',
+            'principal_investigator_id': 3,
+            'publication_doi': ['10.100/123456', '10.100/7891011'],
+            'study_title':
+                'Identification of the Microbiomes for Cannabis Soils',
+            'metadata_complete': True,
+            'number_samples_collected': 27,
+            'study_abstract':
+                'This is a preliminary study to examine the microbiota '
+                'associated with the Cannabis plant. Soils samples '
+                'from the bulk soil, soil associated with the roots, '
+                'and the rhizosphere were extracted and the DNA '
+                'sequenced. Roots from three independent plants of '
+                'different strains were examined. These roots were '
+                'obtained November 11, 2011 from plants that had been '
+                'harvested in the summer. Future studies will attempt '
+                'to analyze the soils and rhizospheres from the same '
+                'location at different time points in the plant '
+                'lifecycle.'
+            }
+        obs = _build_single_study_info(Study(1), study_info, study_proc,
+                                       proc_samples)
+        self.assertItemsEqual(obs, self.single_exp)
+
+    def test_build_single_proc_data_info(self):
+        obs = _build_single_proc_data_info(4, '18S',
+                                           self.proc_data_exp[0]['samples'])
+        self.assertItemsEqual(obs, self.proc_data_exp[0])
+
+    def test_build_study_info(self):
```

Comment:
I actually decided to do it the other way around - I removed the test below b/c they're doing the same thing, but his one has the assertRaises and does not modify the DB, which allow us to save un DB reset